contracts/strategies/strategy.sol.example

```diff
@@ -0,0 +1,125 @@
+        /// @dev The contract will require one balanceOf() call and one if() block per token
+        /// @dev Example: the JoeAvaxSavax contract has 4 tokens; AVAX, sAVAX, QI, and JOE
+        uint256 _token1 = IERC20(token1).balanceOf(address(this));
+        
+        if (_token1 > 0) {
+            uint256 _keep = _token1.mul(keep).div(keepMax);
+            if (_keep > 0){
+                /// @dev Check the imported base contract and strategy-base.sol contracts for the appropriate
+                /// @dev fee function for the token. Tokens may reqire a specific path to exchange for SNOB
+                _takeFeeWavaxToSnob(_keep);
+            }
+            
+            _token1 = IERC20(wavax).balanceOf(address(this));
+        }
+
+        /// @dev Swap base pair rewards for the other token
+        /// @dev In the case of TOKEN1 Rewards, swap half TOKEN1 for TOKEN2
+        if(_token1 > 0){
+            IERC20(token1).safeApprove(joeRouter, 0);
+            IERC20(token1).safeApprove(joeRouter, _token1.div(2)); 
+            /// @dev  Use the swap function appropriate to your platform. Future swaps will route through our dex aggregator
+            _swapTraderJoe(token1, token2, _token1.div(2)); 
+        }
+
+
+        /// @dev In the case of REWARD1 Rewards, swap REWARD1 for TOKEN1 and TOKEN2
+        if(_reward1 > 0){
+            IERC20(reward1).safeApprove(joeRouter, 0);
+            IERC20(reward1).safeApprove(joeRouter, _reward1);   
+            _swapTraderJoe(reward1, token1, _reward1.div(2));
+            _swapTraderJoe(reward1, token2, _reward1.div(2));
+        }
+
+        
+        /// @dev Add liquidity for TOKEN1/TOKEN2
+        _token1 = IERC20(token1).balanceOf(address(this));
+        _token2 = IERC20(token2).balanceOf(address(this));
+        if (_token1 > 0 && _token2 > 0) {
+            IERC20(token1).safeApprove(joeRouter, 0);
+            IERC20(token1).safeApprove(joeRouter, _token1);
+
+            IERC20(token2).safeApprove(joeRouter, 0);
+            IERC20(token2).safeApprove(joeRouter, _token2);
+
+            IJoeRouter(joeRouter).addLiquidity(
+                token1,
+                token2,
+                _token1,
+                _token2,
+                0,
+                0,
+                address(this),
+                now + 60
+            );
+
+            /// @dev Just as with the fee and swap functions above, one balanceOf() and one if() block reqired per token
+            /// @dev Check balances and donate dust to the treasury
+            _token1 = IERC20(token1).balanceOf(address(this));
+            
+            if (_token1 > 0){
+                IERC20(token1).transfer(
+                    IController(controller).treasury(),
+                    _token1
+                );
+            }
+
+             
+        }
+
+        _distributePerformanceFeesAndDeposit();
+    }
+
+    /// @dev The contract name on line 123 must match the contract name on line 10
```

Comment:
Maybe add an explanation why they must match, i.e. for deployment automation script or something.